### PR TITLE
Mutator Refactor

### PR DESCRIPTION
**Pull request by BenTheKush**
This PR is currently for staging. In this PR I:
1. refactor mutant generation to produce all mutants deterministically
2. make several code cleanliness refactors

### CODE REVIEW

File: src/ast.rs [hunk 1]
```diff
@@ -32,10 +32,10 @@ struct Replacement {
...
 
 /// Solidity AST representation.
-/// There are two fields, `element`
-/// which is the underlying json object representing
-/// an AST node and `contract` which indicates
-/// the name of the contract that this node belongs to.
+///
+/// There are two fields, `element` which is the underlying json object
+/// representing an AST node and `contract` which indicates the name of the
+/// contract that this node belongs to.
```
Comment (BenTheKush) [inline src/ast.rs:38]: Defaulting to 80-char lines, hope this is okay: it's blowing up diffs a bit, but I think it makes comments more readable

File: src/mutant_writer.rs [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+            &filename.display()
+        );
+
+        fs::create_dir_all(filename.parent().unwrap())?;
+        fs::write(filename.as_path(), mutant_contents)?;
+
+        Ok(filename)
+    }
+
+    /// Get the filename where a Mutant will be exported to. This depends
```
Comment (chandrakananandi) [inline src/mutant_writer.rs:158]: incomplete comment?

File: src/mutant_writer.rs [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+    fn diff_mutant(mutant: &Mutant) -> Result<String, Box<dyn error::Error>> {
+        let orig_contents: String = String::from_utf8_lossy(mutant.source.contents()).into();
+        let mutant_contents = mutant.as_source_file().unwrap();
+
+        let diff = TextDiff::from_lines(&orig_contents, &mutant_contents)
+            .unified_diff()
+            .header("original", "mutant")
+            .to_string();
+        // Do writing here.
```
Comment (chandrakananandi) [inline src/mutant_writer.rs:180]: was this supposed to happen?

**Review by chandrakananandi — COMMENTED** [review]
Amazing work! thanks for the PR! Left some comments.

File: .vscode/launch.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
+{
```
[the remark this answers is not in the document]
Comment (BenTheKush) [reply .vscode/launch.json:1]: :+1:

File: Cargo.toml [hunk 5]
```diff
@@ -1,24 +1,31 @@
 [package]
 name = "gambit"
-version = "0.1.0"
+version = "0.2.0"
```
[the remark this answers is not in the document]
Comment (BenTheKush) [reply Cargo.toml:3]: Good question: I think there are some API changes/etc, so I was going for a 0.2...we are breaking backwards compatability, but I honestly don't know best practices here! I'll leave it up to you!

**Review by chandrakananandi — APPROVED** [review]
Still running some tests but looks good!!

**Review by stp59 — APPROVED** [review]
Really exciting to see all of these changes landing!

**Pull request merged**